ojd_daps_skills/pipeline/skill_ner/combine_labels.py

```diff
@@ -112,11 +118,90 @@ def combine_results(labelled_data_s3_folders, keep_id_dict, metadata_jobids):
                 job_labels[job_id] = {
                     "text": job_advert_labels["task"]["data"]["text"],
                     "labels": job_advert_labels["result"],
+                    "type": "label-studio",
                 }
 
     return job_labels
 
 
+def load_format_prodigy(prodigy_labelled_data_s3_folders):
+    """
+    Load all prodigy labels
+    Since these were labelled in 5 sentence chunks, then sort them into a nested dict
+    with the job advert id and the sentence chunk number
+
+    """
+    s3 = get_s3_resource()
+    prodigy_data_chunks = defaultdict(dict)
+    for prodigy_labelled_data_s3_folder in prodigy_labelled_data_s3_folders:
+        prodigy_data = load_prodigy_jsonl_s3_data(
+            s3, bucket_name, prodigy_labelled_data_s3_folder
+        )
+        for p in prodigy_data:
+            if p["answer"] == "accept":
+                prodigy_data_chunks[str(p["meta"]["id"])][p["meta"]["chunk"]] = p
+    return prodigy_data_chunks
+
+
+def combine_prodigy_spans(prodigy_data_chunks):
+    """
```

Comment:
this is such a pain for prodigy/spacy more generally - when dealing with start/end char indexes in spacy Span's, they have a nice parameter called `alignment_mode` which allows you to define how character indices snap to token boundaries i.e. Options: "strict" (no snapping), "contract" (span of all tokens completely within the character span), "expand" (span of all tokens at least partially covered by the character span). Defaults to "strict". This is what i modified when I needed to deal with spans for the company descriptions recipe. 

More info in the [Span.char_span method in the Span documentation ](https://spacy.io/api/span). Not sure if this is totally relevant here but something to consider.

Reply:
oh that's good to know about! thanks 🙏